packages/database/src/core/RepoManager.ts

```diff
@@ -107,6 +102,11 @@ export class RepoManager {
     const parsedUrl = parseRepoInfo(dbUrl);
     const repoInfo = parsedUrl.repoInfo;
 
+    let dbEmulatorHost = process.env[FIREBASE_DATABASE_EMULATOR_HOST_VAR];
+    if (dbEmulatorHost) {
+      dbUrl = `http://${dbEmulatorHost}?ns=${parseURL(dbUrl).subdomain}`;
```

Comment:
What if `dbUrl` is already an Emulator URL, like `http://localhost:9000/?ns=foo`? For example, I have a Node.js script that is already configured to write to the emulator but I still want to run it through `emulator:exec` (which will set the env var).

Reply:
In this case the result will be a no-op: https://github.com/firebase/firebase-js-sdk/blob/6b53e0058483c9002d2fe56119f86fc9fb96b56c/packages/database/src/core/util/libs/parser.ts#L185

Reply:
Good stuff!

Reply:
Can we replace `parseURL(dbUrl).subdomain` with `repoInfo.namespace` ?  Seems like we're parsing twice in two different ways...